presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/QueryBuilder.java

```diff
@@ -285,6 +282,33 @@ private String toPredicate(String columnName, String operator, Object value, Jdb
         return quote(columnName) + " " + operator + " ?";
     }
 
+    private String getGroupBy(Optional<List<List<JdbcColumnHandle>>> groupingSets)
+    {
+        if (groupingSets.isEmpty()) {
+            return "";
+        }
+
+        verify(!groupingSets.get().isEmpty());
+        if (groupingSets.get().size() == 1) {
+            List<JdbcColumnHandle> groupingSet = getOnlyElement(groupingSets.get());
+            if (groupingSet.isEmpty()) {
+                // global aggregation
+                return "";
+            }
+            return " GROUP BY " + groupingSet.stream()
+                    .map(JdbcColumnHandle::getColumnName)
+                    .map(this::quote)
+                    .collect(joining(", "));
+        }
+        return " GROUP BY GROUPING SETS " +
```

Comment:
This could be handled differently in different sql engines

Reply:
We can revisit as needed, wdyt?